[ENH] JudgeLog service(updateOrInsert, find) 작성

kimhanui opened the pull request:
## 한 것
- updateOrInsert, find 작성
  - updateOrInsert는 트랙판정할 때 실행되는 메서드
  - find메서드는 아직 컨트롤러와 연결은 ❌ 
- 좋은 테코는 아니지만 테코도 작성

## 질문
https://github.com/riyenas0925/Sejong_Track_Management/issues/213#issuecomment-612461840

## 앞으로 todo
  - ApiTrackJudgeController -> pnp결과 찾아야됨
  - delete 작성

kimhanui committed:
[ENH] JudgeLog service(updateOrInsert, find) 작성

- todo
  - ApiTrackJudgeController -> pnp결과 찾아야됨
  - delete 작성

kimhanui committed:
MERGE branch 'develop' of https://github.com/riyenas0925/Sejong_Track_Management into feature/issue-213

riyenas0925 commented on src/main/java/kr/ac/sejong/domain/trackJudge/JudgeLog/JudgeLogCustomRepository.java:
```diff
@@ -0,0 +1,16 @@
+
+    public List<JudgeLog> findAllByTrack(Long trackId);
+
+    public Optional<JudgeLog> findByMemberAndTrack(String userId, Long trackId);
```
@kimhanui 여기서 다른곳은 List인데 여기만 Optional인 이유가 있음??

kimhanui replied:
@riyenas0925 여기만 최소 한 엔티티만 확인하면 됨(그래서 querydsl에서도 fetchOne씀) +  Optinal쓰는게 null값 들어가도(찾는 엔티티가 없을 때)npe안 뜨고 좋더라고

riyenas0925 commented on src/main/java/kr/ac/sejong/service/JudgeLogService.java:
```diff
@@ -0,0 +1,96 @@
+    @Transactional(readOnly = true)
+    public List<JudgeLog> findByMember(String userId) throws Exception {
+        return repo.findAllByMember(userId);
+    }
```
@kimhanui 이부분도 엔티티인 JudgeLog를 반환하지 말고 ResponseDto 하나 만들어서 반환하자

kimhanui replied:
@riyenas0925 ㅇㅋㅇㅋ

riyenas0925 commented on src/main/java/kr/ac/sejong/service/JudgeLogService.java:
```diff
@@ -0,0 +1,96 @@
+            log.info("judgeLog update fail ! and being inserted.....");
+            insert(dto);
+        }
+    }
```
@kimhanui 여기서 updateByMember로 따로 뺀이유가 있음?

kimhanui replied:
@riyenas0925 insert나 delete는 memeber, track 양쪽에서 관리할 수 있는데 update는 member만 관리함 + 메서드명이나 parameter 보고 로직을 바로 알아보기엔 parameter가 다 JudgeLogDto로 똑같음. -> 한번봣을 때 헷갈리지 않으려면 명시해주는게 좋겠다고 생각햇음ㅋㅋ

kimhanui committed:
[CLN] JudgeLog service 수정

- updateOrInsert만 남겨두고 update, insert문은 삭제
   => 비즈니스적 로직인거지, 일어나선 안되는 예외가 아니므로 try-catch을 대체하기 위함
   => 자연스럽게 tx도 updateOrInsert에만 남겨둠

- find메서드에서 dto 반환

- 부족한 점: Exception을 구분 안 하고 생각 없이 던졌다.(service/controller둘다)
